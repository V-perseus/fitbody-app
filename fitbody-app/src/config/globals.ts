import { Dimensions, TextProps } from 'react-native'
import styles from './styles'
import string from './strings'
import { FB_CURRENT_ENV, FB_ENVIRONMENTS } from '../../env'

export const globals = {
  // Styles & Colors
  styles: {
    ...styles,
  },
  // Fonts
  fonts: {
    ...styles.fonts,
  },
  // Header Styles
  header: {
    icons: {
      marginRight: 15,
      marginLeft: 15,
    },
    headerTitleStyle: <TextProps>{
      fontSize: 16,
      fontFamily: styles.fonts.primary.semiboldStyle.fontFamily,
      textAlign: 'center',
    },
    headerTruncatedBackTitle: true,
  },
  // Standard String
  strings: {
    ...string,
  },
  // Background Images
  bgImage: {
    imageStyle: { resizeMode: 'cover' },
    style: {
      width: Dimensions.get('window').width,
      height: Dimensions.get('window').height,
    },
  },
  // Window dimensions
  window: {
    width: Dimensions.get('window').width,
    height: Dimensions.get('window').height,
  },
  // Button click targeting
  touchableOpacity: {
    activeOpacity: 0.8,
    hitSlop: { top: 5, right: 5, left: 5, bottom: 5 },
  },
  fbEnvironments: {
    ...FB_ENVIRONMENTS,
  },
  currentEnv: FB_CURRENT_ENV,
  get apiBase(): string {
    return this.fbEnvironments[this.currentEnv]
  },
}

export default globals
