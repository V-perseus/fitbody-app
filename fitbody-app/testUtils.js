import React from 'react'
import { render as rtlRender } from '@testing-library/react-native'
import { configureStore } from '@reduxjs/toolkit'
import { NavigationContainer } from '@react-navigation/native'
import { createStackNavigator } from '@react-navigation/stack'
import { Provider } from 'react-redux'

// Import your own reducer
import { appReducer } from './src/store'
import { ModalStackScreens } from './src/config/routes/loggedInStack'

// custom render method override
function render(ui, { preloadedState, store = configureStore({ reducer: appReducer, preloadedState }), ...renderOptions } = {}) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>
  }
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions })
}

// re-export everything
export * from '@testing-library/react-native'
// override render method
export { render }

// custom navigation mock
const Stack = createStackNavigator()
export const MockedNavigator = ({
  component,
  params = {},
  preloadedState,
  store = configureStore({ reducer: appReducer, preloadedState }),
}) => {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="MockedScreen" component={component} initialParams={params} />
          <Stack.Screen name="Modals" component={ModalStackScreens} />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  )
}
