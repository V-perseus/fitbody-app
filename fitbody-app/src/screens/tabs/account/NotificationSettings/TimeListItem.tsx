import React from 'react'
import { StyleSheet, View, Text } from 'react-native'
import moment from 'moment'

import TimeIcon from '../../../../../assets/images/svg/icon/24px/time.svg'
import TrashIcon from '../../../../../assets/images/svg/icon/16px/trash.svg'

import globals from '../../../../config/globals'
import { ButtonOpacity } from '../../../../components/Buttons/ButtonOpacity'

interface ITileListItemProps {
  onPress: () => void
  time: string
}
export const TimeListItem: React.FC<ITileListItemProps> = ({ onPress, time }) => {
  return (
    <View style={styles.container}>
      <View style={styles.rowCenter}>
        <TimeIcon color={globals.styles.colors.colorWhite} />
        <Text style={styles.text}>{moment().startOf('day').add(moment.duration(time)).format('hh:mm A')}</Text>
      </View>
      <ButtonOpacity onPress={onPress} testID="del_button">
        <TrashIcon color={globals.styles.colors.colorWhite} />
      </ButtonOpacity>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    borderRadius: 8,
    paddingHorizontal: 15,
    paddingVertical: 7,
    marginBottom: 7,
    backgroundColor: globals.styles.colors.colorSkyBlue,
  },
  rowCenter: { flexDirection: 'row', alignItems: 'center' },
  text: { marginLeft: 15, fontWeight: 'bold', color: globals.styles.colors.colorWhite },
})
