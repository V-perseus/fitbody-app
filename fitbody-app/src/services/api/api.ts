import * as Sentry from '@sentry/react-native'
import deviceInfoModule from 'react-native-device-info'
import { Platform } from 'react-native'

// Services
import { setErrorMessage } from '../error'
import { displayLoadingModal, hideLoadingModal } from '../loading'
import NavigationService from '../NavigationService'
import globals from '../../config/globals'
import { updateUserIsRestricted } from '../../data/user'
import { storeRegistry } from '../../store/storeRegistry'

const API_BASE = globals.apiBase

type Methods = 'get' | 'post' | 'patch' | 'put' | 'delete'
type MethodsUppercase = 'GET' | 'POST' | 'PATCH' | 'PUT' | 'DELETE'
interface IDefaultHeaders {
  [key: string]: string
}
interface IParams {
  method: MethodsUppercase
  headers: IDefaultHeaders
  body?: FormData | string
}
export interface IAppErrorResponse {
  response: {
    data: {
      message: string
    }
  }
}
export function isAppErrorResponse(error: any): error is IAppErrorResponse {
  return error.response?.data?.message !== undefined
}

export const apiPublic = <T>(url: string, method: Methods = 'get', data: {} | null = {}, loader = true, headers = {}): Promise<T> => {
  // console.log('api',`${API_BASE}${url}`);
  if (loader) {
    displayLoadingModal()
  }

  let token = storeRegistry.getState().services.session.apiToken
  let defaultHeaders: IDefaultHeaders = {
    'Content-Type': 'application/json',
    Accept: 'application/json',
    'X-App-Version': deviceInfoModule.getVersion(),
    'X-App-Os': Platform.OS,
  }
  if (token) {
    defaultHeaders.Authorization = 'Bearer ' + token
  }

  let reqHeaders = {}
  Object.assign(reqHeaders, defaultHeaders)
  let params: IParams = {
    method: method.toUpperCase() as MethodsUppercase,
    headers: reqHeaders,
  }

  if ((method.toUpperCase() === 'POST' || method.toUpperCase() === 'PUT' || method.toUpperCase() === 'PATCH') && data) {
    if (data instanceof FormData) {
      params.body = data
    } else {
      params.body = JSON.stringify(data)
    }
  }

  let req = new Request(API_BASE + url, params)

  return fetch(req)
    .then((response) => {
      if (response.ok) {
        if (loader) {
          hideLoadingModal()
        }
        let responseText = response.text()
        return responseText
          .then(function (text) {
            return text ? Promise.resolve(JSON.parse(text)) : Promise.resolve({})
          })
          .catch(function (error) {
            return Promise.reject(error)
          })
      }
      if (loader) {
        hideLoadingModal()
      }
      return Promise.reject(response)
    })
    .catch((error: Response) => {
      let message: string | Error = ''
      if (error && error instanceof Error) {
        try {
          Sentry.captureException(new Error(error.message || JSON.stringify(error)))
        } catch {}
        message = error
      } else if (error && error.json) {
        error.json().then((errorJson) => {
          message = errorJson.message
          if (message && typeof message === 'string' && message.toLowerCase().includes('please purchase')) {
            updateUserIsRestricted(true)
            NavigationService.sendToPaywall()
          } else if (!url.includes('/v4/events')) {
            setErrorMessage({ error: message })
          }
        })
      }
      hideLoadingModal()
      return Promise.reject(message)
    })
}

/**
 * Sign a user in
 */
export const apiAuth = <T>(url: string, method: Methods = 'get', data: null | {} = {}, loader = true, headers = {}): Promise<T> => {
  return apiPublic<T>(url, method, data, loader, headers)
}
