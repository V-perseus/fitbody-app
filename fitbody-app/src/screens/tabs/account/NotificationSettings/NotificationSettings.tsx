import React, { useEffect, useLayoutEffect, useState } from 'react'
import { View, Text, TouchableOpacity, ScrollView, Platform, Switch, StatusBar, StyleSheet } from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'
import moment from 'moment'
import { NavigationProp } from '@react-navigation/native'

// Components
import TimePickerModal from '../../../../components/TimePickerModal'
import { TimeListItem } from './TimeListItem'

// Styles and Assets
import globals from '../../../../config/globals'
import TimeIcon from '../../../../../assets/images/svg/icon/24px/time.svg'
import ChevronRight from '../../../../../assets/images/svg/icon/16px/cheveron/right.svg'

// Services and Data
import api from '../../../../services/api'

// Types
import { AccountStackParamList } from '../../../../config/routes/routeTypes'
import { INotificationSetting, IUpdateNotificationSettingPayload } from '../../../../data/notification/types'

type SettingTimes = Record<string, string[]>

interface INotificationSettingsProps {
  navigation: NavigationProp<AccountStackParamList, 'NotificationSettings'>
}
const NotificationSettings: React.FC<INotificationSettingsProps> = ({ navigation }) => {
  const [settings, setSettings] = useState<INotificationSetting[]>([])
  const [settingsLookup, setSettingsLookup] = useState<Record<string, INotificationSetting>>({})
  const [values, setValues] = useState<Record<string, boolean>>({})
  const [times, setTimes] = useState<SettingTimes>({})
  const [modalOpen, setModalOpen] = useState(false)
  const [activeSetting, setActiveSetting] = useState<number | null>(null)

  const save = (s: Record<string, INotificationSetting>, v: Record<string, boolean>, t: SettingTimes) => {
    const persist: IUpdateNotificationSettingPayload = {}

    for (let k of Object.keys(s)) {
      const newk = k.split('_')[1]
      if (v[k] !== undefined && v[k] !== s[k].settings.enabled) {
        persist[newk] = {
          enabled: v[k],
        }
      }

      if (s[k].type === 'timeSelector') {
        if (persist[newk] === undefined && t[k] !== undefined) {
          persist[newk] = { enabled: s[k].settings.enabled }
        }

        if (persist[newk] !== undefined) {
          persist[newk].times = t[k] !== undefined ? t[k] : s[k].settings.times
        }
      }
    }

    api.notifications.saveSettings({ settings: persist }).then(() => navigation.goBack())
  }

  useLayoutEffect(() => {
    navigation.setOptions({
      headerTitle: () => <Text style={globals.header.headerTitleStyle}>NOTIFICATIONS</Text>,
      headerLeft: () => (
        <TouchableOpacity onPress={() => navigation.goBack()} style={globals.header.icons}>
          <View style={styles.headerButton}>
            <Text style={styles.headerButtonText}>CANCEL</Text>
          </View>
        </TouchableOpacity>
      ),
      headerRight: () => (
        <TouchableOpacity onPress={() => save(settingsLookup, values, times)} style={globals.header.icons}>
          <View style={styles.headerButton}>
            <Text style={styles.headerButtonText}>SAVE</Text>
          </View>
        </TouchableOpacity>
      ),
    })
  }, [navigation, values, times, settingsLookup])

  useEffect(() => {
    api.notifications
      .settings()
      .then((d) => {
        setSettings(d.notificationTypes)
        const flattened = flattenSettings(d.notificationTypes[0], {})
        setSettingsLookup(flattened)
      })
      .catch(() => {
        // console.log(error)
      })
  }, [])

  const flattenSettings = (setting: INotificationSetting, o: Record<string, INotificationSetting>) => {
    o[`setting_${setting.id}`] = setting

    for (const child of setting.children || []) {
      flattenSettings(child, o)
    }
    return o
  }

  const toggleSetting = (setting: INotificationSetting, newValue: boolean) => {
    let newObject = { ...values, [`setting_${setting.id}`]: newValue }
    if (setting.cascading_mode === 'inherit' && setting.id !== 1) {
      setting.children.forEach((child) => {
        newObject = { ...newObject, [`setting_${child.id}`]: newValue }
      })
    }

    setValues(newObject)
  }

  const addTimeToSetting = (setting: INotificationSetting) => {
    setActiveSetting(setting.id)
    if (times[`setting_${setting.id}`] === undefined) {
      setTimes({ ...times, [`setting_${setting.id}`]: setting.settings.times! })
    }
    setModalOpen(true)
  }

  const addTime = (time: string) => {
    let newObject = {
      ...times,
      [`setting_${activeSetting}`]: [...new Set([...(times[`setting_${activeSetting}`] || []), time])].sort((a, b) => {
        return moment(a, 'hh:mm a').isBefore(moment(b, 'hh:mm a')) ? -1 : 1
      }),
    }

    setModalOpen(false)
    setTimes(newObject)
  }

  const removeTime = (setting: INotificationSetting, index: number) => {
    let existing: string[] | undefined
    if (times[`setting_${setting.id}`] === undefined) {
      existing = setting.settings.times
    } else {
      existing = times[`setting_${setting.id}`]
    }

    if (existing !== undefined) {
      existing.splice(index, 1)
      const newObj = { ...times, [`setting_${setting.id}`]: existing }

      setTimes(newObj)
    }
  }

  const renderSettingOrGroup = (setting: INotificationSetting, parentValue: boolean, indent = 0) => {
    const isPaused = settings[0].settings.enabled === false || values.setting_1 === false ? true : false

    const isEnabled = values[`setting_${setting.id}`] !== undefined ? values[`setting_${setting.id}`] : setting.settings.enabled

    const isGroup = (setting.children && setting.children.length > 0 && indent > 0) || (indent == 1 && setting.type === 'timeSelector')

    return (
      <View
        key={setting.id}
        style={{ flex: 1, borderTopWidth: isGroup ? 7 : 0, borderTopColor: `rgba(250,141,213,${isPaused ? 0.5 : 1})` }}>
        <View
          style={{
            flexDirection: 'row',
            alignItems: 'center',
            height: 50,
            paddingHorizontal: 33,
            justifyContent: 'space-between',
            flex: 1,
            borderBottomWidth: isGroup && setting.type !== 'timeSelector' ? 2 : 0,
            borderBottomColor: globals.styles.colors.colorGray,
          }}>
          <Text
            style={{
              fontWeight: 'bold',
              opacity: isPaused ? 0.5 : 1,
              fontSize: isGroup ? 16 : 14,
              color: isGroup ? globals.styles.colors.colorPink : globals.styles.colors.colorBlack,
            }}>
            {isGroup ? setting.title.toUpperCase() : setting.title}
          </Text>
          {(setting.type === 'toggle' || setting.type === 'timeSelector') && (
            <Switch
              testID={setting.title}
              thumbColor={globals.styles.colors.colorWhite}
              disabled={isPaused && indent !== 0}
              value={indent === 0 ? !isEnabled : isEnabled}
              onValueChange={(v) => toggleSetting(setting, indent === 0 ? !v : v)}
              trackColor={{ true: globals.styles.colors.colorLove, false: globals.styles.colors.colorGrayLight }}
            />
          )}
        </View>
        {setting.type === 'timeSelector' && (
          <View style={{ paddingHorizontal: 33, flex: 1 }}>
            <View
              style={{
                borderBottomWidth: 1,
                alignItems: 'center',
                marginBottom: 15,
                borderBottomColor: `rgba(229, 229, 299, ${isEnabled ? '1' : '0.5'})`,
              }}>
              <View style={{ marginBottom: -9, paddingHorizontal: 10, backgroundColor: globals.styles.colors.colorWhite }}>
                <Text style={{ color: globals.styles.colors.colorGrayDark, opacity: isEnabled ? 1 : 0.5 }}>Remind me at these times:</Text>
              </View>
            </View>
            {(times[`setting_${setting.id}`] !== undefined ? times[`setting_${setting.id}`] : setting.settings.times || []).map((t, i) => (
              <TimeListItem key={`${setting.id}_${i}`} time={t} onPress={isEnabled ? () => removeTime(setting, i) : () => {}} />
            ))}
            <TouchableOpacity onPress={isEnabled ? () => addTimeToSetting(setting) : () => {}}>
              <View style={styles.addTimeButton}>
                <View style={styles.rowCenter}>
                  <TimeIcon color={isEnabled && !isPaused ? globals.styles.colors.colorGrayDark : globals.styles.colors.colorGray} />
                  <Text
                    style={{
                      marginLeft: 15,
                      fontWeight: '600',
                      color: isEnabled && !isPaused ? globals.styles.colors.colorGrayDark : globals.styles.colors.colorGray,
                    }}>
                    Add Time
                  </Text>
                </View>
                <ChevronRight color={isEnabled && !isPaused ? globals.styles.colors.colorGrayDark : globals.styles.colors.colorGray} />
              </View>
            </TouchableOpacity>
          </View>
        )}
        {setting.children.map((element) =>
          renderSettingOrGroup(
            element,
            parentValue === false ? false : setting.cascading_mode === 'disable' ? isEnabled : true,
            indent + 1,
          ),
        )}
      </View>
    )
  }

  return (
    <SafeAreaView style={[globals.styles.container, { backgroundColor: globals.styles.colors.colorWhite }]}>
      <ScrollView style={{ flex: 1, marginTop: Platform.OS === 'ios' ? 50 : 50 + (StatusBar.currentHeight || 0) }}>
        {settings[0] ? renderSettingOrGroup(settings[0], true) : null}
      </ScrollView>
      <TimePickerModal testID="add_time_modal" onChange={(time) => addTime(time)} onClose={() => setModalOpen(false)} visible={modalOpen} />
    </SafeAreaView>
  )
}

const styles = StyleSheet.create({
  rowCenter: { flexDirection: 'row', alignItems: 'center' },
  addTimeButton: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
    borderRadius: 8,
    paddingHorizontal: 15,
    paddingVertical: 7,
    marginBottom: 7,
    backgroundColor: globals.styles.colors.colorGrayLight,
  },
  headerButton: {
    borderRadius: 12,
    backgroundColor: globals.styles.colors.colorWhite,
    borderStyle: 'solid',
    borderWidth: 1,
    borderColor: globals.styles.colors.colorBlack,
    width: 60,
  },
  headerButtonText: {
    fontFamily: globals.fonts.primary.style.fontFamily,
    paddingVertical: 3,
    fontSize: 11,
    textAlign: 'center',
    color: globals.styles.colors.colorBlack,
  },
})

export default NotificationSettings
