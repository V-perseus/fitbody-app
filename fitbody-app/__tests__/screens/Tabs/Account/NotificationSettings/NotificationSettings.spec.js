import { fireEvent, MockedNavigator, render, act, cleanup } from '../../../../../testUtils'

import NotificationSettings from '../../../../../src/screens/tabs/account/NotificationSettings'
import api from '../../../../../src/services/api'
import globals from '../../../../../src/config/globals'

jest.mock('../../../../../src/services/api')

describe('NotificationSettings screen', () => {
  let component = null
  let setting = null
  beforeEach(() => {
    api.notifications.settings.mockResolvedValue(setting)
    component = MockedNavigator({ component: NotificationSettings })
    setting = {
      notificationTypes: [
        {
          id: 1,
          class: '',
          title: 'Pause All Notifications',
          type: 'toggle',
          cascading_mode: 'inherit',
          default: { enabled: true },
          settings: { enabled: true },
          children: [
            {
              id: 2,
              class: '',
              title: 'General',
              type: 'toggle',
              cascading_mode: 'inherit',
              default: { enabled: true },
              settings: { enabled: true },
              children: [
                {
                  id: 6,
                  class: 'ProgressPhoto',
                  title: 'Progress Photos',
                  type: 'toggle',
                  cascading_mode: 'inherit',
                  default: { enabled: true },
                  settings: { enabled: true },
                  children: [],
                },
              ],
            },
            {
              id: 4,
              class: 'WaterIntake',
              title: 'Water Intake',
              type: 'timeSelector',
              cascading_mode: 'inherit',
              default: { enabled: true, times: ['10:00'] },
              settings: { enabled: true, times: ['7:00', '7:30', '8:30', '9:00'] },
              children: [],
            },
            {
              id: 7,
              class: '',
              title: 'Workouts',
              type: 'toggle',
              cascading_mode: 'inherit',
              default: { enabled: true },
              settings: { enabled: true },
              children: [
                {
                  id: 8,
                  class: 'DailyWorkout',
                  title: 'Daily Workout Reminder',
                  type: 'timeSelector',
                  cascading_mode: 'inherit',
                  default: { enabled: true, times: ['8:00'] },
                  settings: { enabled: true, times: ['8:00'] },
                  children: [],
                },
                {
                  id: 9,
                  class: 'NewChallenge',
                  title: 'New Cardio Burn Workouts',
                  type: 'toggle',
                  cascading_mode: 'inherit',
                  default: { enabled: true },
                  settings: { enabled: true },
                  children: [],
                },
              ],
            },
          ],
        },
      ],
    }
  })
  afterEach(() => {
    cleanup()
  })

  it('renders snapshot correctly', async () => {
    await act(async () => {
      await api.notifications.settings.mockResolvedValue(setting)
      const tree = render(component).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  it('should render header buttons', async () => {
    const tree = render(component)
    await act(async () => {
      const { getByText, findByText } = tree
      await findByText('NOTIFICATIONS')
      getByText(/cancel/gi)
      getByText(/save/gi)
    })
  })

  it('should render toggle', async () => {
    const tree = render(component)
    await act(async () => {
      const { findByText, findByTestId } = tree
      await findByText('NOTIFICATIONS')
      const toggle = await findByTestId(/Water Intake/i)
      expect(toggle).not.toBeNull()
      expect(toggle).toHaveProp('trackColor', {
        false: globals.styles.colors.colorGrayLight,
        true: globals.styles.colors.colorLove,
      })
      fireEvent(toggle, 'valueChange', { value: true })
    })
  })

  it('should render available settings', async () => {
    const tree = render(component)
    await act(async () => {
      const { getAllByText, getByText, findByText } = tree
      await findByText('NOTIFICATIONS')
      getByText(/Pause All Notifications/gi)
      getByText(/Progress Photos/gi)
      getByText(/Daily Workout Reminder/gi)
      getByText(/New Cardio Burn Workouts/gi)
      getByText(/07:30/i)
      getByText(/07:00/i)
      getByText(/08:30/i)
      expect(getAllByText(/add time/i)).toHaveLength(2)
    })
  })

  it('should open Add Time modal', async () => {
    const tree = render(component)
    await act(async () => {
      const { getAllByText, findByText, findByTestId } = tree
      await findByText('NOTIFICATIONS')
      const buttons = getAllByText(/add time/i)
      expect(buttons).toHaveLength(2)
      fireEvent.press(buttons[0])
      const modal = await findByTestId('add_time_modal')
      expect(modal).not.toBeNull()
    })
  })
})
