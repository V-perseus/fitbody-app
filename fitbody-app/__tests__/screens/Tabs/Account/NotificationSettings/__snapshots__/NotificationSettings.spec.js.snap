// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationSettings screen renders snapshot correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "zIndex": 1,
          }
        }
      >
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          onLayout={[Function]}
          pointerEvents="box-none"
          style={null}
        >
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "opacity": 1,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "zIndex": 0,
              }
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "rgb(255, 255, 255)",
                  "borderBottomColor": "rgb(216, 216, 216)",
                  "flex": 1,
                  "shadowColor": "rgb(216, 216, 216)",
                  "shadowOffset": Object {
                    "height": 0.5,
                    "width": 0,
                  },
                  "shadowOpacity": 0.85,
                  "shadowRadius": 0,
                }
              }
            />
          </View>
          <View
            collapsable={false}
            pointerEvents="box-none"
            style={
              Object {
                "height": 44,
                "maxHeight": undefined,
                "minHeight": undefined,
                "opacity": undefined,
                "transform": undefined,
              }
            }
          >
            <View
              pointerEvents="none"
              style={
                Object {
                  "height": 0,
                }
              }
            />
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "alignItems": "stretch",
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-start",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginStart": 0,
                    "opacity": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "marginLeft": 15,
                      "marginRight": 15,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#ffffff",
                        "borderColor": "#333333",
                        "borderRadius": 12,
                        "borderStyle": "solid",
                        "borderWidth": 1,
                        "width": 60,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#333333",
                          "fontFamily": "OpenSans",
                          "fontSize": 11,
                          "paddingVertical": 3,
                          "textAlign": "center",
                        }
                      }
                    >
                      CANCEL
                    </Text>
                  </View>
                </View>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "justifyContent": "center",
                    "marginHorizontal": 16,
                    "maxWidth": 590,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    Object {
                      "fontFamily": "OpenSans-Semibold",
                      "fontSize": 16,
                      "textAlign": "center",
                    }
                  }
                >
                  NOTIFICATIONS
                </Text>
              </View>
              <View
                collapsable={false}
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "flex-end",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "justifyContent": "center",
                    "marginEnd": 0,
                    "opacity": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  collapsable={false}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "marginLeft": 15,
                      "marginRight": 15,
                      "opacity": 1,
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "backgroundColor": "#ffffff",
                        "borderColor": "#333333",
                        "borderRadius": 12,
                        "borderStyle": "solid",
                        "borderWidth": 1,
                        "width": 60,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#333333",
                          "fontFamily": "OpenSans",
                          "fontSize": 11,
                          "paddingVertical": 3,
                          "textAlign": "center",
                        }
                      }
                    >
                      SAVE
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          gestureResponseDistance={
            Object {
              "bottom": -1,
              "end": -1,
              "start": -1,
              "top": -1,
            }
          }
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                handlerTag={1}
                handlerType="PanGestureHandler"
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <RNCSafeAreaView
                        edges={
                          Array [
                            "bottom",
                            "left",
                            "right",
                            "top",
                          ]
                        }
                        style={
                          Array [
                            Object {
                              "flex": 1,
                            },
                            Object {
                              "backgroundColor": "#ffffff",
                            },
                          ]
                        }
                      >
                        <RCTScrollView
                          style={
                            Object {
                              "flex": 1,
                              "marginTop": 50,
                            }
                          }
                        >
                          <View />
                        </RCTScrollView>
                        <Modal
                          animationType="none"
                          deviceHeight={null}
                          deviceWidth={null}
                          hardwareAccelerated={false}
                          hideModalContentWhileAnimating={false}
                          onBackdropPress={[Function]}
                          onModalHide={[Function]}
                          onModalWillHide={[Function]}
                          onModalWillShow={[Function]}
                          onRequestClose={[Function]}
                          panResponderThreshold={4}
                          scrollHorizontal={false}
                          scrollOffset={0}
                          scrollOffsetMax={0}
                          scrollTo={null}
                          statusBarTranslucent={false}
                          supportedOrientations={
                            Array [
                              "portrait",
                              "landscape",
                            ]
                          }
                          swipeThreshold={100}
                          transparent={true}
                          visible={false}
                        />
                      </RNCSafeAreaView>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
